orion/pkg/repos/program_repo.go

```diff
@@ -163,26 +163,34 @@ func (pr *programRepo) Insert(program domains.Program) error {
 	return handleSqlExecResult(execResult, 1, "program was not inserted")
 }
 
-func (pr *programRepo) Publish(programIds []string) []domains.PublishErrorBody {
+func (pr *programRepo) Publish(programIds []string) ([]domains.PublishErrorBody, error) {
 	errorList := make([]domains.PublishErrorBody, 0)
 
+	tx, err := pr.db.Begin()
+	if err != nil {
+		return nil, err
+	}
+	stmt, err := tx.Prepare("UPDATE programs SET published_at=? WHERE program_id=? AND published_at IS NULL")
+	if err != nil {
+		return nil, err
+	}
+	defer stmt.Close()
```

Comment:
In addition to specific errors per program id, the function can also return general errors that don't pertain to a specific id (i.e. tx.Prepare statement fails). I'm not sure if returning two items is the best way to deal with this.
The alternative is to concatenate the errors immediately on the repo level instead of waiting until later, but then the PublishErrorBody struct becomes obsolete. I think that might be better though. What do yall think?

Reply:
I did the latter. The PublishErrorBody isn't obsolete because it removes the need to create a separate struct for each one.

Reply:
I'm saying we don't need any struct at all if we do it the latter way.